framework/sandstone.h

```diff
@@ -493,12 +529,62 @@ static inline void *aligned_alloc_safe(size_t alignment, size_t size)
     return aligned_alloc(alignment, aligned_size);
 }
 
+/**
+ * @brief If compatibility mode has been requested
+ *
+ * It is allowed to follow old pattern of random number (exact number of bits consumed from
+ * RNG each time), to execute the tests with exactly the same data with previous (pre-100)
+ * suites. For compatibility mode old set_random_bits() is used, otherwise mantissa of floats
+ * is composed from 0s/1s chains (of particular lengths)
+ *
+ * @return true if compatibility mode was requested
+ * @return false to follow new patterns of random data
+ */
+bool random_compatitibilitiy_mode();
+
 /// Returns a 64 bit unsigned integer in which num_bits_to_set of the
 /// first bitwidth bits are randomly set.  For example,
 /// set_random_bits(2, 8) would return a uint64_t in which 2 of the
 /// least significant 8 bits are randomly set and all other bits are 0.
 uint64_t set_random_bits(unsigned num_bits_to_set, uint32_t bitwidth);
 
+/** RNG state */
+typedef struct {
+    uint8_t bits_available;
+    uint32_t random_bits;
+} random_bits_state_t;
+
+/**
+ * @brief Get specified number of bits from RNG
+ *
+ * @param bits  number of bits to fetch (and to "consume" from RNG)
+ * @param state RNG state
+ * @return uint64_t
+ */
+uint64_t random_bits(uint8_t bits, random_bits_state_t* state);
+
+/// macro to find most significant bit in u16 value
+/// Expected to be used in compile time expressions
+#define _POW2_BIT(n, b) ((n) & (1 << ((b) - 1))) ? (b) :
```

Comment:
Nitpick: underscore + capital is reserved for the compiler. I suggest moving the underscore to the end.

Same question as above: do tests need this or can this be in sandstone_p.h or just random.cpp?